core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeDDLBuilder.scala

```diff
@@ -219,4 +218,38 @@ class SnowflakeDDLBuilder
       affectedColumns.map(col => ir.DropConstraint(Some(col), constraint))
     }
   }
+
+  private def buildVariables(ctx: DeclareContext): Seq[ir.SetVariable] = {
+    val ids = ctx.id().asScala.map(_.getText)
+    val dataTypes = ctx.dataType().asScala.map(dt => Some(DataTypeBuilder.buildDataType(dt)))
+    val expressions = ctx.expr().asScala.map(_.accept(expressionBuilder)).map(Some(_))
+
+    // Padding the lists to ensure they all have the same size
+    // [TODO]: This is a temporary fix. We need to handle this in a better way
+    val paddedDataTypes = dataTypes ++ Seq.fill(ids.size - dataTypes.size)(None)
+    val paddedExpressions = (expressions ++ Seq.fill(ids.size - expressions.size)(None))
+
+    ids.zip(paddedDataTypes).zip(paddedExpressions).map { case ((name, dataType), expr) =>
+      ir.SetVariable(name, dataType, expr)
+    }
+
+  }
+
+  private def buildReturn(ctx: ReturnStatementContext): ir.Expression = {
+    // [TODO]: implement return statement for TABLE(res)
+    ctx.accept(expressionBuilder)
+  }
```

Comment:
There's nothing wrong with this method _per se_, but we usually use this pattern of defining a `private def buildXYZ(...): XYZ` only when `XYZ` isn't a type returned by an existing `SnowflakeXYZBuilder`. Otherwise, we simply delegate to said `SnowflakeXYZBuilder`.

Reply:
Done